### ISSUE TITLE
Autocomplete issue

### ISSUE THREAD

**Issue by Kennys-Kardboard-Box** [godotengine/godot#66672]
### Godot version

4.0 beta1

### System information

Windows 11

### Issue description

when entering data into a function, the autocomplete only shows the base function and does not allow user to autocomplete variable names.
![godot_4_autocomplete_bug](https://user-images.githubusercontent.com/99441282/193320509-fd469cf3-22b4-4d07-85e7-38cc5dae485e.png)

### Steps to reproduce

use a built-in function. start typing in variable name in function. Moving away and back to function does not work.

**Comment by DragonAxe** [comment]
I have been experiencing this as well. I can reproduce this all the way back in Alpha1 and as recent as a 2022-09-27 build of `master`.

Here is a minimum working script. Create a new script with the following contents.
```
extends Node

var my_special_little_variable

func _process(delta: float) -> void:
    clamp()
```
Now attempt to autocomplete (Ctrl+Space) `my_special_little_variable` within the `clamp` statement.

#### Expected:
A popup (or two) with both function parameters and autocomplete suggestions. Below is a mock up image I made with both autocomplete and function parameter hint bubbles.
![image](https://user-images.githubusercontent.com/12703200/193395878-4b016039-eee1-45ab-b99c-08f22be68616.png)

#### Actual:
Function parameters are shown, but pressing Ctrl+Space does not bring up autocomplete until the cursor is outside the function's parenthesis.
![image](https://user-images.githubusercontent.com/12703200/193395782-c85f79e2-00b8-4657-9cb7-9bb261579d87.png)

System:
Linux 5.15.65-1-MANJARO

**Comment by Mickeon** [comment]
I can confirm autocompletion has essentially been dysfunctional for a while. Really difficult to work with.

**Comment by mattazz** [comment]
I notice that when referring to other nodes, the auto complete doesn't show the methods or variables in that node.

``` $AnimatedSprite2D```
 does not show completion for ".play()"

**Comment by PoisonousGame** [comment]
I can confirm that custom variables, functions are not autocomplete, and many built-in functions are not available, such as print(),randf_range()... And autocomplete sorting is not perfect, common functions are not first at the top. I think the 3.5 autocomplete is really fine.

**Comment by PoisonousGame** [comment]
godot4 beta2 ,I found that after declaring a variable or defining a function you need to save it, Only then autocomplete will show it. Autocomplete commonly used functions are not sorted at the top, 3.5 does it very well.

**Comment by EricEzaM** [comment]
I cannot reproduce. Could you try latest master? I could also not reproduce `I found that after declaring a variable or defining a function you need to save it, Only then autocomplete will show it`.

https://user-images.githubusercontent.com/41730826/194313859-cac40cfe-b598-4dbe-95c5-63c528c4db07.mp4

I do think that the function signature hint should stay visible though - or at least, there should be a way to make it pop up again, like Ctrl+Shift+Space @Paulb23. See a nice example from visual studio

https://user-images.githubusercontent.com/41730826/194315101-eb8af798-e242-4077-ba36-bd673be081c9.mp4

**Comment by Kakiharu** [comment]
I too encountered this in beta 1 and figured I'd wait out for beta 2 but alas the bug is still there. Its like once you  start typing the code completion forgets anything once you get inside to entering variables. When I first encountered it I was translating a Godot 3.5 project and and had to fix rotation_degrees into rotation() but once I started to type deg for deg2rad I was confused because I had no autocompletion, I first thought was no way they removed deg_to_rad. Hopefully this gets squashed soon. Would love to start learning Godot 4 but I do not enjoy looking at the manual every time to look up a function. 

TLDR; tried to do rotation(deg_to_rad) but autocomplete stops suggestions once inside the rotation function.

**Comment by midiphony** [comment]
> I notice that when referring to other nodes, the auto complete doesn't show the methods or variables in that node.
> 
> ` $AnimatedSprite2D` does not show completion for ".play()"

-- There is this issue, about autocompletion not working when referring relative nodes :
https://github.com/godotengine/godot/issues/62797

**Comment by KoBeWi** [comment]
I just encountered a case that I think is the same issue (or related):

https://github.com/godotengine/godot/assets/2223172/f72a7426-052b-4b44-a208-4b67ad1b4ef5

Autocompletion for `input_vector` local variable does not show if trying to complete inside a function call and with `.` on the right. It works without `.` or outside function.

Code for testing:
```GDScript
var input_vector := Input.get_vector(&"ui_left", &"ui_right", &"ui_up", &"ui_down")

velocity = input_vector * 300
move_and_slide()

if absf(input_vector.x) < 0.1 or absf(input_vector.x) < 0.1:
```

**Comment by HolonProduction** [comment]
Can't reproduce in v4.4.dev.custom_build [5675c7646] (both the original issue and KoBeWi's addition).
